pkg/podgrouper/podgrouper/plugins/jobset/jobset_grouper.go

```diff
@@ -0,0 +1,148 @@
+// Copyright 2025 NVIDIA CORPORATION
+// SPDX-License-Identifier: Apache-2.0
+
+package jobset
+
+import (
+	"fmt"
+	"math"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/NVIDIA/KAI-scheduler/pkg/podgrouper/podgroup"
+	"github.com/NVIDIA/KAI-scheduler/pkg/podgrouper/podgrouper/plugins/defaultgrouper"
+)
+
+const (
+	// JobSet pod labels (duplicated to avoid importing JobSet project).
+	jobSetLabelJobSetName        = "jobset.sigs.k8s.io/jobset-name"
+	jobSetLabelReplicatedJobName = "jobset.sigs.k8s.io/replicatedjob-name"
+	jobSetPodGroupNamePrefix     = "pg"
+)
+
+// JobSetGrouper creates a PodGroup per replicatedJob for JobSet workloads.
+// PodGroup name: pg-<jobset-name>-<replicatedjob-name>-<jobset-uid>
+// MinAvailable: replicas * min(parallelism, completions if set) (defaults to 1).
+type JobSetGrouper struct {
+	*defaultgrouper.DefaultGrouper
+}
+
+func NewJobSetGrouper(defaultGrouper *defaultgrouper.DefaultGrouper) *JobSetGrouper {
+	return &JobSetGrouper{
+		DefaultGrouper: defaultGrouper,
+	}
+}
+
+func (g *JobSetGrouper) Name() string {
+	return "JobSet Grouper"
+}
+
+// +kubebuilder:rbac:groups=jobset.x-k8s.io,resources=jobsets,verbs=get;list;watch
+// +kubebuilder:rbac:groups=jobset.x-k8s.io,resources=jobsets/finalizers,verbs=patch;update;create
+
+func (g *JobSetGrouper) GetPodGroupMetadata(
+	topOwner *unstructured.Unstructured,
+	pod *v1.Pod,
+	_ ...*metav1.PartialObjectMetadata,
+) (*podgroup.Metadata, error) {
+	pgMeta, err := g.DefaultGrouper.GetPodGroupMetadata(topOwner, pod)
+	if err != nil {
+		return nil, err
+	}
+
+	jobSetName := topOwner.GetName()
+	jobSetUID := topOwner.GetUID()
+	if jobSetName == "" || len(jobSetUID) == 0 {
+		return nil, fmt.Errorf("jobset top owner %s/%s missing name or UID", topOwner.GetNamespace(), topOwner.GetName())
+	}
+
+	replicatedJobName, ok := pod.Labels[jobSetLabelReplicatedJobName]
+	if !ok || replicatedJobName == "" {
+		return nil, fmt.Errorf("pod %s/%s missing required label %q", pod.Namespace, pod.Name, jobSetLabelReplicatedJobName)
+	}
+
+	pgMeta.Name = fmt.Sprintf(
+		"%s-%s-%s-%s",
+		jobSetPodGroupNamePrefix,
+		jobSetName,
+		replicatedJobName,
+		string(jobSetUID),
```

Comment:
Maybe a switch between the replicatedJobName and uid would look better? Then it's easier to understand that the uid belong to the jobset and not the replicatedJob.